engine/tests/phpunit/Elgg/Application/ServeFileHandlerTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Elgg\Application;
+
+use PHPUnit_Framework_TestCase;
+
+class ServeFileHandlerTest extends PHPUnit_Framework_TestCase {
+
+	/**
+	 * @var ServeFileHandler
+	 */
+	protected $handler;
+
+	/**
+	 * @var \ElggFile
+	 */
+	protected $file;
+
+	public function setUp() {
+		$app = _elgg_testing_application();
+		$dataroot = _elgg_testing_config()->getDataPath();
+
+		$session = \ElggSession::getMock();
+		_elgg_services()->setValue('session', $session);
+		_elgg_services()->session->start();
+
+		$site_secret_mock = $this->getMockBuilder('\Elgg\Database\SiteSecret')->getMock();
+		$site_secret_mock->method('init')->willReturn('strongSiteSecret1234567890');
+		_elgg_services()->setValue('siteSecret', $site_secret_mock);
+
+		$this->handler = new ServeFileHandler($app);
+
+		$file_mock = $this->getMockBuilder('\ElggFile')->disableOriginalConstructor()->getMock();
+		$file_mock->method('getFileNameOnFilestore')->willReturn("{$dataroot}file_service/foobar.txt");
+		$file_mock->method('exists')->willReturn(true);
+		$this->file = $file_mock;
+	}
+
+	function createRequest($file) {
+		$site_url = elgg_get_site_url();
+		$url = $file->getURL();
+		$path = substr($url, strlen($site_url));
+		$path_key = \Elgg\Application::GET_PATH_KEY;
+		return \Elgg\Http\Request::create("?$path_key=$path");
+	}
+
+	/**
+	 * @group FileService
+	 */
+	public function testSend400WhenURLIsMalformmatted() {
+
+		$request = \Elgg\Http\Request::create('');
+		$response = $this->handler->getResponse($request);
+		$this->assertEquals(400, $response->getStatusCode());
+	}
+
+	/**
+	 * @group FileService
+	 */
+	public function testSend403WhenURLExpired() {
+
+		$file = new \Elgg\FileService\File();
+		$file->setFile($this->file);
+		$file->setExpires('-1 day');
+
+		$request = $this->createRequest($file);
+		$response = $this->handler->getResponse($request);
+		$this->assertEquals(403, $response->getStatusCode());
+	}
+
+	/**
+	 * @group FileService
+	 */
+	public function testResponseWhenFileModifiedTime() {
```

Comment:
Test names should describe the expected result. E.g. `testSends403onFileModificationTimeMismatch` or similar.